topics/single-cell/tutorials/alevin-commandline/tutorial.md

```diff
@@ -0,0 +1,677 @@
+| N704 | 3    | wildtype    | male    |
+| N705 | 4    | wildtype    | male    |
+| N706 | 5    | wildtype    | male    |
+| N707 | 6    | knockout    | male    |
+
+We are currently analysing sample N701, so let's add its information from the table. 
+
+## Batch 
+
+We will label batch as an integer - "0" for sample N701, "1" for N702 etc. The way to do it is creating a list with zeros of the length corresponding to the number of cells that we have in our SummarizedExperiment object, like so:
+
+```bash
+batch <- rep("0", length(colnames(alevin_se)))
+```
+
+And now create a batch slot in the *colData names* and append the `batch` list in the same way as we did with barcodes:
+
+```bash
+colData(alevin_se)$batch <- batch
+colData(alevin_se)
+```
+
+A new column appeared, full of zeros - as expected! 
+
+## Genotype
+
+It's all the same for genotype, but instead creating a list with zeros, we'll create a list with string "wildtype" and append it into genotype slot:
+
+```bash
+genotype <- rep("wildtype", length(colnames(alevin_se)))
+colData(alevin_se)$genotype <- genotype
+```
+
+## Sex 
+
+You already know what to do, right? A list with string "male" and then adding it into a new slot! 
+```bash
+sex <- rep("male", length(colnames(alevin_se)))
+colData(alevin_se)$sex <- sex
+```
+
+Check if all looks fine:
+```bash
+colData(alevin_se)
+```
+
+3 new columns appeared with the information that we've just added - perfect! You can add any information you need in this way, as long as it's the same for all the cells from one sample. 
+
+ 
+# Adding gene metadata 
+
+The genes IDs are stored in *rownames*. Let's exctract them into a separate object:
+
+```bash
+gene_ID <- rownames(alevin_se)
+```
+
+Analogically, we will add those genes IDs into *rowData names* which stores gene metadata. To do this, we will create a column called `gene_ID` in *rowData* and pass the stored values into there.
+
+```bash
+rowData(alevin_se)$gene_ID <- gene_ID
+```
+
+## Adding genes symbols based on their IDs
+
+Since gene symbols are much more informative than only gene IDs, we will add them to our metadata. We will base this annotation on Ensembl - the genome database – with the use of the library BioMart. We will use the archive Genome assembly GRCm38 to get the gene names. Please note that the updated version (GRCm39) is available, but some of the gene IDs are not in that EnsEMBL database. The code below is written in a way that it will work for the updated dataset too, but will produce ‘NA’ where the corresponding gene name couldn’t be found.
+
+```bash
+# get relevant gene names
+library("biomaRt")                                      # load the BioMart library
+ensembl.ids <- gene_ID                               
+mart <- useEnsembl(biomart = "ENSEMBL_MART_ENSEMBL")    # connect to a specified BioMart database and dataset hosted by Ensembl
+ensembl_m = useMart("ensembl", dataset="mmusculus_gene_ensembl", host='https://nov2020.archive.ensembl.org') 	
```

Comment:
Use `version=100` instead of `host`. It is also easy to justify that we use version 100 because our GTF and FASTA are from version 100.